### ISSUE TITLE
Select Distinct with where/Any subquery always includes key from joined table, and returns too many rows

### ISSUE THREAD

**Issue by JasonBurkey** [aspnet/EntityFramework#5630]
### Steps to reproduce

There are 3 very simple tables in this scenario.  You could have 3 tables with a GUID and a string, and that would work fine (GUID is PKEY).  These represent a Drive/Folder/File type situation.  I am attempting to find Drives that have files.

I call the following:
``` C#
(from root in rootSet join folder in folderSet on root.StorageRootId equals folder.RootId
where fileSet.Any(file => file.FolderId == folder.FolderId)
select root.StorageRootId).Distinct();
```

This produces the following query:

``` SQL
SELECT DISTINCT [folder].[FolderId], [root].[StorageRootId]
FROM [StorageRoot] AS [root]
INNER JOIN [StorageFolder] AS [folder] ON [root].[StorageRootId] = [folder].[RootId]
WHERE EXISTS (
    SELECT 1
    FROM [StorageFile] AS [file]
    WHERE [file].[FolderId] = [folder].[FolderId])
```

And too many rows are returned because the FolderID is returned.  I wish to only have Roots.

If the where clause is removed, then the select ONLY returns the StorageRootId

This query:
``` C#
(from root in rootSet join folder in folderSet on root.StorageRootId equals folder.RootId
select root.StorageRootId).Distinct();
```

Produces this SQL:

``` SQL
SELECT DISTINCT [root].[StorageRootId]
FROM [StorageRoot] AS [root]
INNER JOIN [StorageFolder] AS [folder] ON [root].[StorageRootId] = [folder].[RootId]
```

You can see, the FolderId is not returned this time, and it shouldn't be since I explicitly select just 1 column

### The issue
The first query should not be returning the folder.FolderId column, and is causing an incorrect result set.

This only occurs when the where / Any clause is present.

```
Exception message: None
Stack trace: N/A
```

### Further technical details

EF Core version: 1.0.0-rc2-final
Operating system: n/a

Other details about my project setup:

**Comment by divega** [comment]
@maumar can you please investigate?

**Comment by maumar** [comment]
simplified repro:
```cs
    class Program
    {
        static void Main(string[] args)
        {
            using (var ctx = new MyContext())
            {
                ctx.Database.EnsureCreated();
                var query = (from root in ctx.Roots
                              where ctx.Folders.Any(f => f.Id == root.Id)
                              select root.Name).Distinct();

                var result = query.ToList();
            }
        }
    }

    public class Root
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public List<Folder> Folders { get; set; }
    }

    public class Folder
    {
        public int Id { get; set; }
        public int RootId { get; set; }
        public Root Root { get; set; }
    }

    public class MyContext : DbContext
    {
        public DbSet<Root> Roots { get; set; }
        public DbSet<Folder> Folders { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer(new SqlConnectionStringBuilder { DataSource = ".", InitialCatalog = "Repro5630", MultipleActiveResultSets = true, IntegratedSecurity = true }.ToString());
        }
    }
```

produces:
```sql
SELECT DISTINCT [root].[Id], [root].[Name]
FROM [Roots] AS [root]
WHERE EXISTS (
    SELECT 1
    FROM [Folders] AS [f]
    WHERE [f].[Id] = [root].[Id])
```

**Comment by maumar** [comment]
fixed in c25acc9a72b003680430ccf7854db07d5dfba7c7
